massdriver/resource_artifact_test.go

```diff
@@ -23,14 +23,45 @@ func TestAccMassdriverArtifactBasic(t *testing.T) {
 	})
 }
 
+func TestAccMassdriverArtifactRename(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckMassdriverArtifactConfigBasic(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMassdriverArtifactExists("massdriver_artifact.new"),
+				),
+			},
+			{
+				Config: testAccCheckMassdriverArtifactConfigResourceNameChange(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMassdriverArtifactExists("massdriver_artifact.new_name"),
+				),
+			},
+		},
+	})
+}
+
 func testAccCheckMassdriverArtifactConfigBasic() string {
 	return `
 	resource "massdriver_artifact" "new" {
 		field = "example-artifact"
 		provider_resource_id = "arn:::something"
 		type = "type"
-		schema_path = "testdata/schema-artifacts.json"
```

Comment:
How are these tests working without these being set?

Reply:
Great question! I'm still downloading `python` and _then_ `localstack`. I want to run the ACC tests before merging as well.

Reply:
I yanked those because I was looking at the [metadata struct](https://github.com/massdriver-cloud/terraform-provider-massdriver/blob/main/massdriver/resource_artifact.go#L20-L23) and not the schema def, which is what I wanted. I thought maybe we yanked them as fields once we set the defaults. Thanks for catching this and saving me some time later 💪